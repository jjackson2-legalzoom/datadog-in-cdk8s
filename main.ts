import { Construct } from 'constructs';
import { App, Chart, ChartProps } from 'cdk8s';
import { BasicObservabilityBundle } from './lib/datadog';

export class BasicObservabilityBundleDemoChart extends Chart {
  constructor(scope: Construct, id: string, props: ChartProps = { }) {
    super(scope, id, props);

    new BasicObservabilityBundle(this, 'basic-observability-bundle', {
      serviceName: 'cartservice',
      warningThreshold: 99,
      targetThreshold: 90
    })

  }
}

const app = new App();
new BasicObservabilityBundleDemoChart(app, 'datadog-in-cdk8s');
app.synth();
