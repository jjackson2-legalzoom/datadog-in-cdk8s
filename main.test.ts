import {BasicObservabilityBundleDemoChart} from './main';
import {Testing} from 'cdk8s';

describe('Placeholder', () => {
  test('Empty', () => {
    const app = Testing.app();
    const chart = new BasicObservabilityBundleDemoChart(app, 'test-chart');
    const results = Testing.synth(chart)
    expect(results).toMatchSnapshot();
  });
});
