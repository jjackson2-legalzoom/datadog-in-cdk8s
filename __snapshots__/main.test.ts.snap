// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder Empty 1`] = `
Array [
  Object {
    "apiVersion": "datadog.upbound.io/v1alpha1",
    "kind": "DashboardJSON",
    "metadata": Object {
      "name": "test-chart-dashboard-c8f9124c",
    },
    "spec": Object {
      "forProvider": Object {
        "dashboard": "{\\"title\\":\\"jackjack-cdk8s-demo Basic Observability Bundle Dashboard for cartservice\\",\\"widgets\\":[],\\"template_variables\\":[],\\"layout_type\\":\\"ordered\\",\\"notify_list\\":[],\\"reflow_type\\":\\"fixed\\"}",
      },
      "providerConfigRef": Object {
        "name": "datadog-provider-config",
      },
    },
  },
]
`;
